Plugins/GitUIPluginInterfaces/GitUIPluginInterfaces.csproj

```diff
@@ -9,6 +9,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="CommunityToolkit.Mvvm" />
```

Comment:
Could you please help me understand what this package brings?

Reply:
The CommunityToolkit.Mvvm provides a framework for the Model-View-ViewModel pattern.
It generates the tedious boilerplate code parts using Roslyn.
And it provides a message bus (`IMessenger`, `IRecipient<TMessage>`).

Alright, my first implementation jumped too short.
The fixup commit adds a `RepoModifiedMessage` and provides the `GitRepoViewModel` using the Service Registry.
The changes to `FormBrowse.cs`, `ScriptRunner.cs` and `RevisionGridControl.cs` contain the actual bugfix and demonstrate the usage.

The decoupling by means of the `RepoModifiedMessage` facilitates the implementation #11273.

I have prepared MVVM for the imlementation of the GitOutputControl.
I am going to use MVVM for the implementation of the FileStatusList using TreeView.

Reply:
> The CommunityToolkit.Mvvm provides a framework for the Model-View-ViewModel pattern.

I would generally discourage this as the MVVM is an anti-pattern in my experience. Avoiding MVVM brings the following benefits:

1. Simplified testing
2. Simplified code through the removal of unnecessary abstraction layers
3. Improved performance within WPF

Reply:
> 1. Simplified testing

Separation of concerns simplifies testing. Please do not tell me ViewModels were more difficult to test than FormBrowse and ROT!

Reply:
@mstv MVVM certainly makes testing more complicated in practice. The original claims that it would simplify testing were flawed because they assumed that testing could not be accurately performed against the view, and therefore any code which was part of the view would permanently be untested.

In practice, MVVM always makes testing slightly more complicated:

1. All of the tests of the original view are still required
2. Tests of the model are still required
3. If (1) and (2) do not cover the view model completely, more tests need to be added for the view model

In cases where (1) was elided in favor of (3), I've seen those cases be prone to user-visible product regressions. With tests of the view being the most accurate and important of the UI tests, and those tests always being able to cover the view logic, it's rare for MVVM to bring anything beneficial to the table for testing scenarios.